python_modules/libraries/dagster-dg/dagster_dg/cli/component.py

```diff
@@ -310,88 +315,101 @@ def component_check_command(
-            top_level_errs = list(
-                top_level_component_validator.iter_errors(component_doc_tree.value)
-            )
-            for err in top_level_errs:
-                validation_errors.append(ErrorInput(None, err, component_doc_tree))
-            if top_level_errs:
-                continue
-
-            component_key = ComponentKey.from_typename(
-                component_doc_tree.value.get("type"), dirpath=component_path.parent
-            )
-            component_contents_by_key[component_key] = component_doc_tree
-            if isinstance(component_key, LocalComponentKey):
-                local_component_dirs.add(component_dir)
-
-    # Fetch the local component types, if we need any local components
-    component_registry = RemoteComponentRegistry.from_dg_context(
-        dg_context, local_component_type_dirs=list(local_component_dirs)
-    )
-    for component_key, component_doc_tree in component_contents_by_key.items():
-        try:
-            json_schema = component_registry.get(component_key).component_params_schema or {}
-
-            v = Draft202012Validator(json_schema)
-            for err in v.iter_errors(component_doc_tree.value["params"]):
-                validation_errors.append(ErrorInput(component_key, err, component_doc_tree))
-        except KeyError:
-            # No matching component type found
-            validation_errors.append(
-                ErrorInput(
-                    None,
-                    ValidationError(
-                        f"Component type '{component_key.to_typename()}' not found in {component_key.python_file}."
-                        if isinstance(component_key, LocalComponentKey)
-                        else f"Component type '{component_key.to_typename()}' not found."
-                    ),
-                    component_doc_tree,
-                )
-            )
-    if validation_errors:
-        for component_key, error, component_doc_tree in validation_errors:
-            click.echo(
-                error_dict_to_formatted_error(
-                    component_key,
-                    error,
-                    source_position_tree=component_doc_tree.source_position_tree,
-                    prefix=["params"] if component_key else [],
+        if validation_errors:
+            for component_key, error, component_doc_tree in validation_errors:
+                click.echo(
+                    error_dict_to_formatted_error(
+                        component_key,
+                        error,
+                        source_position_tree=component_doc_tree.source_position_tree,
+                        prefix=["params"] if component_key else [],
+                    )
                 )
-            )
-        context.exit(1)
+            return False
+        else:
+            click.echo("All components validated successfully.")
+            return True
+
+    if watch:
+        watched_paths = (
```

Comment:
should we maybe output something here to indicate that its in this mode?

Reply:
the watch utils do that for us:

```bash

/Users/ben/repos/components_demo/dagster-workspace/projects/sql-transform/sql_transform/defs/my_sql/component.yaml:3 - sql_transform.lib.SqlComponent attributes 'sql_path' is a required property
     | 
   1 | type: sql_transform.lib.SqlComponent
   2 | 
   3 | attributes:
     | ^ 'sql_path' is a required property
   4 |   sql_pdath: my_sql.sql
   5 |   sql_engine_url: 'f'
   6 |   asset_specs:
     | 


Updated at 10:34:44, watching for changes...
```